### DISCUSSION TITLE
Railsやるときのフロントエンド構成どうしてます？

### DISCUSSION THREAD

**Discussion by danimal141** [roppongirb/discussions#25] [Ideas]

**Comment by yusukesugasawa** [comment]
https://github.com/roppongirb/discussions/issues/24

**Comment by yusukesugasawa** [comment]
石井
Marionette.js使ってた
そこからReact

いまは、ReactはSPAにして、RailsでAPIたてる感じ

**Comment by yusukesugasawa** [comment]
大倉さん

Raact×TSは結構多い
いろんな使い方ある

結局管理画面はRailsで書くのが、、、、

脱webpackerが気になってる

**Comment by yusukesugasawa** [comment]
田渕さん

がっつりフロントエンド大事かどうかできってる

がっつり必要なら
next.jsで

がっつりいらないならRailsから？

webpackerでesbuild？（あってる？）というのがいい感じ

**Reply by okuramasafumi** [reply]
https://github.com/rails/dartsass-rails
SASSの機能が使えない件、DHHがgemを作っていますね。

**Reply by hatappi** [reply]
https://esbuild.github.io/

**Reply by yusukesugasawa** [reply]
https://esbuild.github.io/

**Reply by tabuchik** [reply]
Railsでコンポーネント指向ができるようになってきている

https://viewcomponent.org/
だったり、gemじゃなくても公式にRenderable objectというのが導入された

**Reply by tabuchik** [reply]
esbuildを使う場合にはwebpackerではなくwebpackですね(自分の場合は)

**Reply by okuramasafumi** [reply]
https://github.com/rails/jsbundling-rails
これもあります。

**Comment by yusukesugasawa** [comment]
榊原さん

ユーザーニーズのあるところはAngularで

Raact採用したいとかはあるけど、
いまはRailsからslimで書いてるのでどうしよう！なところ

**Comment by yusukesugasawa** [comment]
濱田さん

最初は、jQueryとcoffeeScriptで書いてた

管理画面問題はある！

**Comment by yusukesugasawa** [comment]
大原さん

Rails5のときはAssetpipelineから
Reactに乗っかっていきたい

Next.jsも多い

そうなるとバックもNode.jsに統一したい！とか出てくる

**Comment by yusukesugasawa** [comment]
たどころさん

フィヨルドでやっているのは、
必要なところにVue.jsをつっこむ

既存のプラグインの学習コストが高い、、、！

**Comment by yusukesugasawa** [comment]
eieiさん

railsでAPIにして、Next.jsが多い

管理画面問題
社内系のアプリは、jQueryを直がきしてる
それで必要十分なんで

**Comment by yusukesugasawa** [comment]
はたっぴさん

フロントはReactで、バックをgraphql-ruby

個人だと、
hypernovaつかってた

**Comment by yusukesugasawa** [comment]
深堀りポイント

- Railsとフロントエンド切り分けられている話
  - ケース・バイ・ケース？
- 管理画面話

**Comment by tabuchik** [comment]
管理画面はJSを使わない！という強い気持ちで進めていたことがある
社内向けなら意外となんとかなる

(rails-ujsとCSSフレームワークのみをCDNから読み込む)

**Comment by yusukesugasawa** [comment]
RubyとTypeScript問題（Threadでいい感じに、、、🙏）

**Reply by okuramasafumi** [reply]
結局「TypeScriptとその他問題」では 😆

**Comment by yusukesugasawa** [comment]
GraphQLフロント楽だけど、結局API側に面倒くささ寄ってる問題（Threadでいい感じに、、、🙏）
